test/jdk/java/awt/Dialog/DialogResizeTest.java

```diff
@@ -0,0 +1,134 @@
+                2. For both dialog resizable states (resizable, non-resizable) try to
+                   change the size of the dialog. When isResizable is true the dialog
+                   is resizable. When isResizable is false the dialog is non-resizable
+                3. If this is the behavior that you observe, the test has passed, Press
+                   the Pass button. Otherwise the test has failed, Press the Fail button
+                """;
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(initialize())
+                .build()
+                .awaitAndCheck();
+    }
+
+    public static Dialog initialize() {
+        Frame f = new Frame("Owner Frame");
+        MyDialog ld = new MyDialog(f);
+        ld.setBounds(100, 100, 400, 150);
+        ld.setResizable(false);
+        System.out.println("isResizable is set to: " + ld.isResizable());
+        return ld;
+    }
+}
+
+class MyDialog extends Dialog implements ItemListener, ComponentListener {
+    String sText = "Tests java.awt.Dialog.setResizable method";
+    TextArea ta = new TextArea(sText, 2, 40, TextArea.SCROLLBARS_NONE);
+
+    public MyDialog(Frame f) {
+
+        super(f, "setResizable test", false);
+        this.addComponentListener(this);
+
+        Panel cbPanel = new Panel();
+        cbPanel.setLayout(new FlowLayout());
+
+        Panel taPanel = new Panel();
+        taPanel.setLayout(new FlowLayout());
+        taPanel.add(ta);
+
+        Checkbox cb = new Checkbox("Check this box to change the dialog's " +
+                "resizable state", null, isResizable());
+        cb.addItemListener(this);
+        cbPanel.add(cb);
+
+        add("North", taPanel);
+        add("South", cbPanel);
+        pack();
+    }
+
+    public void itemStateChanged(ItemEvent evt) {
+        setResizable(evt.getStateChange() == ItemEvent.SELECTED);
+
+        boolean bResizeState = isResizable();
+        System.out.println("isResizable is set to: " + bResizeState);
+
+        if (isResizable()) {
+            ta.setText("dialog is resizable (isResizable = " + bResizeState + ")");
+        } else {
+            ta.setText("dialog is NOT resizable (isResizable = " + bResizeState + ")");
+        }
+    }
+
+    public void componentResized(ComponentEvent e) {
+        System.out.println("Dialog width: " + this.getSize().width);
+        System.out.println("Dialog height: " + this.getSize().height);
+    }
+
+    public void componentMoved(ComponentEvent e) {
+        System.out.println("componentMoved event occurred.");
+    }
+
+    public void componentShown(ComponentEvent e) {
+        System.out.println("componentShown event occurred.");
+    }
+
+    public void componentHidden(ComponentEvent e) {
+        System.out.println("componentHidden event occurred.");
+    }
```

Comment:
I guess PassFailJFrame.log would be of more use than System println, if at all these logs are of any help..

Reply:
I'm getting this error if I use ` PassFailJFrame.log`  to `componentResize` method.
`java.lang.NullPointerException: Cannot invoke "javax.swing.JTextArea.append(String)" because "PassFailJFrame.logArea" is null`. Since the test is about resize, I've updated only resize related print statement.

Reply:
Probably component is resized even before PassFailJFrame creates the logArea in Builder interface.
The instructions says 
"When isResizable is true the dialog is resizable. When isResizable is false the dialog is non-resizable" and you are logging "PassFailJFrame.log("isResizable is set to: " + bResizeState);" 

so, it seems ComponentListener is not needed and all these 4 methods can be removed which is only logging and that too is not visible to tester...

Reply:
this is pending..

Reply:
Updated.